{"repository":"https://github.com/eyevinn/channel-engine","dockerfilePath":"Dockerfile","startTime":1672070502913,"clone":{"stdout":"","stderr":"Cloning into '/tmp/dinghy-analysis/eyevinn/channel-engine'...\n","error":null,"commit":"e7389666503c712bc08c02308e01d4865de4047d"},"originalBuild":{"startTimestamp":1672070504578,"endTimestamp":1672070531384,"error":null,"stdout":"Sending build context to Docker daemon  11.13MB\r\r\nStep 1/6 : FROM node:12.16.1-alpine\n12.16.1-alpine: Pulling from library/node\naad63a933944: Already exists\nedd41271d385: Pulling fs layer\ndd731a721451: Pulling fs layer\n495807fcdd37: Pulling fs layer\n495807fcdd37: Download complete\ndd731a721451: Verifying Checksum\ndd731a721451: Download complete\nedd41271d385: Verifying Checksum\nedd41271d385: Download complete\nedd41271d385: Pull complete\ndd731a721451: Pull complete\n495807fcdd37: Pull complete\nDigest: sha256:6b5b783c9cfe229af0bd5b0b677dd32005bb22d58465f3d0fe7fbd1c60ce068c\nStatus: Downloaded newer image for node:12.16.1-alpine\n ---> f77abbe89ac1\nStep 2/6 : WORKDIR /app\n ---> Running in ea571e3d21b5\nRemoving intermediate container ea571e3d21b5\n ---> bb950b7f1f1f\nStep 3/6 : ADD . .\n ---> 630a8b49ef95\nStep 4/6 : RUN npm install\n ---> Running in 136c11ee73f8\n\u001b[91mnpm\u001b[0m\u001b[91m WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!\n\u001b[0m\n> dtrace-provider@0.8.8 install /app/node_modules/dtrace-provider\n> node-gyp rebuild || node suppress-error.js\n\n\u001b[91mgyp\u001b[0m\u001b[91m ERR! find Python \ngyp \u001b[0m\u001b[91mERR! find Python Python is not set from command line or npm configuration\ngyp ERR! \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m Python is not set from environment variable PYTHON\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python checking if \"python\" can be used\ngyp ERR! find Python - \"python\" is not in PATH or produced an error\n\u001b[0m\u001b[91mgyp ERR! find Python checking if \"python2\" can be used\ngyp\u001b[0m\u001b[91m ERR! find Python - \"python2\" is not in PATH or produced an error\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m checking if \"python3\" can be used\ngyp ERR! find Python\u001b[0m\u001b[91m - \"python3\" is not in PATH or produced an error\ngyp ERR! find Python\u001b[0m\u001b[91m \ngyp ERR! find Python **********************************************************\ngyp\u001b[0m\u001b[91m ERR! find Python You need to install the latest version of Python.\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python Node-gyp should be able to find and use Python. If not,\n\u001b[0m\u001b[91mgyp ERR! find Python you can try one of the following options:\ngyp ERR! find Python - Use the switch --python=\"/path/to/pythonexecutable\"\ngyp\u001b[0m\u001b[91m ERR! find Python   (accepted by both node-gyp and npm)\n\u001b[0m\u001b[91mgyp ERR! find Python\u001b[0m\u001b[91m - Set the environment variable PYTHON\ngyp\u001b[0m\u001b[91m ERR! find Python\u001b[0m\u001b[91m - Set the npm configuration variable python:\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python   npm config set python \"/path/to/pythonexecutable\"\ngyp \u001b[0m\u001b[91mERR! find Python\u001b[0m\u001b[91m For more information consult the documentation at:\ngyp \u001b[0m\u001b[91mERR! find Python https://github.com/nodejs/node-gyp#installation\ngyp ERR! \u001b[0m\u001b[91mfind Python **********************************************************\ngyp ERR! find Python \n\u001b[0m\u001b[91mgyp ERR! configure error \n\u001b[0m\u001b[91mgyp ERR! stack Error: Could not find any Python installation to use\ngyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:307:47)\ngyp ERR! \u001b[0m\u001b[91mstack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)\ngyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)\ngyp ERR!\u001b[0m\u001b[91m stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:271:16)\ngyp \u001b[0m\u001b[91mERR! stack     at exithandler (child_process.js:310:5)\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m stack     at ChildProcess.errorhandler (child_process.js:322:5)\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mstack     at ChildProcess.emit (events.js:311:20)\ngyp ERR! stack\u001b[0m\u001b[91m     at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)\ngyp ERR!\u001b[0m\u001b[91m stack\u001b[0m\u001b[91m     at onErrorNT (internal/child_process.js:469:16)\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mstack     at processTicksAndRejections (internal/process/task_queues.js:84:21)\n\u001b[0m\u001b[91mgyp ERR! System Linux 5.4.0-113-generic\ngyp ERR! command\u001b[0m\u001b[91m \"/usr/local/bin/node\" \"/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mcwd /app/node_modules/dtrace-provider\ngyp ERR! node -v v12.16.1\n\u001b[0m\u001b[91mgyp ERR! node-gyp -v\u001b[0m\u001b[91m v5.0.5\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mnot ok \n\u001b[0madded 220 packages from 270 contributors and audited 223 packages in 4.612s\n\n39 packages are looking for funding\n  run `npm fund` for details\n\nfound 3 high severity vulnerabilities\n  run `npm audit fix` to fix them, or `npm audit` for details\nRemoving intermediate container 136c11ee73f8\n ---> 37b36af65b1f\nStep 5/6 : ENV DEBUG=engine*\n ---> Running in 4ff49b8acee8\nRemoving intermediate container 4ff49b8acee8\n ---> 702068c858da\nStep 6/6 : CMD [\"npm\", \"start\"]\n ---> Running in 12becd291c6d\nRemoving intermediate container 12becd291c6d\n ---> 06317e17ad91\nSuccessfully built 06317e17ad91\nSuccessfully tagged channel-engine:latest\n","stderr":"","imageSize":207001771},"repair":{"violations":[{"name":"npmCacheCleanAfterInstall","position":{"start":6,"end":6,"columnStart":4,"columnEnd":15}}],"repairedDockerfile":"FROM node:12.16.1-alpine\n\nWORKDIR /app\n\nADD . .\n\nRUN npm install\nENV DEBUG=engine*\n\nCMD [\"npm\", \"start\"]"},"repairedBuild":{"startTimestamp":1672070531755,"endTimestamp":1672070558120,"error":null,"stdout":"Sending build context to Docker daemon  11.13MB\r\r\nStep 1/6 : FROM node:12.16.1-alpine\n ---> f77abbe89ac1\nStep 2/6 : WORKDIR /app\n ---> Running in 40b53d69b465\nRemoving intermediate container 40b53d69b465\n ---> fc46555e4ece\nStep 3/6 : ADD . .\n ---> 75c4d1cdc0f3\nStep 4/6 : RUN npm install\n ---> Running in b6d0be9b0a4b\n\u001b[91mnpm\u001b[0m\u001b[91m WARN read-shrinkwrap\u001b[0m\u001b[91m This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!\n\u001b[0m\n> dtrace-provider@0.8.8 install /app/node_modules/dtrace-provider\n> node-gyp rebuild || node suppress-error.js\n\n\u001b[91mgyp\u001b[0m\u001b[91m ERR! find Python \ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python Python is not set from command line or npm configuration\ngyp\u001b[0m\u001b[91m ERR! find Python Python is not set from environment variable PYTHON\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR!\u001b[0m\u001b[91m \u001b[0m\u001b[91mfind Python checking if \"python\" can be used\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mfind Python - \"python\" is not in PATH or produced an error\n\u001b[0m\u001b[91mgyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mfind Python\u001b[0m\u001b[91m checking if \"python2\" can be used\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python - \"python2\" is not in PATH or produced an error\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mfind Python checking if \"python3\" can be used\ngyp \u001b[0m\u001b[91mERR! find Python\u001b[0m\u001b[91m - \"python3\" is not in PATH or produced an error\ngyp\u001b[0m\u001b[91m ERR! find Python\u001b[0m\u001b[91m \ngyp \u001b[0m\u001b[91mERR! find Python **********************************************************\ngyp ERR!\u001b[0m\u001b[91m find Python You need to install the latest version of Python.\ngyp \u001b[0m\u001b[91mERR! find Python Node-gyp should be able to find and use Python. If not,\ngyp\u001b[0m\u001b[91m ERR! find Python\u001b[0m\u001b[91m you can try one of the following options:\ngyp \u001b[0m\u001b[91mERR! find Python - Use the switch --python=\"/path/to/pythonexecutable\"\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m find Python\u001b[0m\u001b[91m   (accepted by both node-gyp and npm)\ngyp \u001b[0m\u001b[91mERR! find Python - Set the environment variable PYTHON\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m find Python - Set the npm configuration variable python:\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mfind Python   npm config set python \"/path/to/pythonexecutable\"\ngyp ERR! find Python\u001b[0m\u001b[91m For more information consult the documentation at:\ngyp ERR! \u001b[0m\u001b[91mfind Python https://github.com/nodejs/node-gyp#installation\ngyp ERR!\u001b[0m\u001b[91m find Python **********************************************************\ngyp ERR! \u001b[0m\u001b[91mfind Python \n\u001b[0m\u001b[91mgyp ERR! configure error\u001b[0m\u001b[91m \n\u001b[0m\u001b[91mgyp ERR! stack Error: Could not find any Python installation to use\ngyp \u001b[0m\u001b[91mERR! stack\u001b[0m\u001b[91m     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:307:47)\ngyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)\n\u001b[0m\u001b[91mgyp ERR!\u001b[0m\u001b[91m stack\u001b[0m\u001b[91m     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)\ngyp\u001b[0m\u001b[91m \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mstack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:271:16)\ngyp \u001b[0m\u001b[91mERR! stack     at exithandler (child_process.js:310:5)\ngyp\u001b[0m\u001b[91m ERR! \u001b[0m\u001b[91mstack     at ChildProcess.errorhandler (child_process.js:322:5)\n\u001b[0m\u001b[91mgyp ERR! stack     at ChildProcess.emit (events.js:311:20)\n\u001b[0m\u001b[91mgyp ERR! stack\u001b[0m\u001b[91m     at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)\ngyp \u001b[0m\u001b[91mERR! \u001b[0m\u001b[91mstack     at onErrorNT (internal/child_process.js:469:16)\ngyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m stack     at processTicksAndRejections (internal/process/task_queues.js:84:21)\n\u001b[0m\u001b[91mgyp ERR! System Linux 5.4.0-113-generic\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mcommand \"/usr/local/bin/node\" \"/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"\n\u001b[0m\u001b[91mgyp ERR! \u001b[0m\u001b[91mcwd\u001b[0m\u001b[91m /app/node_modules/dtrace-provider\ngyp\u001b[0m\u001b[91m ERR! node -v v12.16.1\n\u001b[0m\u001b[91mgyp\u001b[0m\u001b[91m ERR!\u001b[0m\u001b[91m node-gyp -v v5.0.5\ngyp ERR!\u001b[0m\u001b[91m not ok\u001b[0m\u001b[91m \n\u001b[0madded 220 packages from 270 contributors and audited 223 packages in 4.394s\n\n39 packages are looking for funding\n  run `npm fund` for details\n\nfound 3 high severity vulnerabilities\n  run `npm audit fix` to fix them, or `npm audit` for details\nRemoving intermediate container b6d0be9b0a4b\n ---> fdab16a05608\nStep 5/6 : ENV DEBUG=engine*\n ---> Running in 94f73915e2c5\nRemoving intermediate container 94f73915e2c5\n ---> 8ec2829a37c9\nStep 6/6 : CMD [\"npm\", \"start\"]\n ---> Running in 4d3a3a15fabc\nRemoving intermediate container 4d3a3a15fabc\n ---> 5c9ac4e6db00\nSuccessfully built 5c9ac4e6db00\nSuccessfully tagged channel-engine:latest\n","stderr":"","imageSize":207001771},"endTime":1672070558480}